django_project/certification/models.py

```diff
@@ -0,0 +1,302 @@
+    )
+
+    email = models.CharField(
+        help_text="Email address",
+        max_length=200,
+        null=False,
+        blank=False
+    )
+
+    slug = models.SlugField(unique=True)
+    objects = models.Manager()
+
+    # noinspection PyClassicStyleClass.
+    class Meta:
+        """ Meta class for Course convener """
+        app_label = 'certification'
+        ordering = ['name']
+
+    def save(self, *args, **kwargs):
+        """ Overloaded save method
+        :param args:
+        :param kwargs:
+        """
+        if not self.pk:
+            words = self.name.split()
+            filtered_words = [t for t in words if t.lower() not in STOP_WORDS]
+            # unidecode() represents special characters (unicode data) in ASCII
+            new_list = unidecode(' '.join(filtered_words))
+            self.slug = slugify(new_list)[:50]
+
+        super(CourseConvener, self).save(*args, **kwargs)
+
+    def __unicode__(self):
+        return u'%s' % self.name
+
+
+class CourseAttendee(models.Model):
+    """ Course Attendee model """
+    name = models.CharField(
+        help_text="First name course attendee",
+        max_length=200,
+        null=False,
+        blank=False
+    )
+
+    surename = models.CharField(
+        help_text="Surename course attendee",
+        max_length=200,
+        null=False,
+        blank=False
+    )
+
+    email = models.CharField(
+        help_text="Email address",
+        max_length=200,
+        null=False,
+        blank=False
+    )
+
+    slug = models.SlugField(unique=True)
+    objects = models.Manager()
+
+    # noinspection PyClassicStyleClass.
+    class Meta:
+        """ Meta class for Course attendee """
+        app_label = 'certification'
+        ordering = ['name']
+
+    def save(self, *args, **kwargs):
+        """ Overloaded save method
+        :param args:
+        :param kwargs:
+        """
+        if not self.pk:
+            words = self.name.split()
+            filtered_words = [t for t in words if t.lower() not in STOP_WORDS]
+            # unidecode() represents special characters (unicode data) in ASCII
+            new_list = unidecode(' '.join(filtered_words))
+            self.slug = slugify(new_list)[:50]
+
+        super(CourseAttendee, self).save(*args, **kwargs)
+
+    def __unicode__(self):
+        return u'%s' % self.name
+
+
+class CertifyingOrganisation(models.Model):
+    name = models.CharField(
+        help_text="Name of Organisation or Institution",
+        max_length=200,
+        null=False,
+        blank=False
+    )
+
+    organisation_email = models.CharField(
+        help_text="Email address Organisation or Institution",
+        max_length=200,
+        null=False,
+        blank=False
+    )
+
+    organisation_phone = models.CharField(
+        help_text="Contact of Organisation or Institution",
+        max_length=200,
+        null=False,
+        blank=False
+    )
+
+    slug = models.SlugField(unique=True)
```

Comment:
you could see a lot of object has slug, and same save method (that generate slug)
you could move these to new object and make it as parent to model that will be used these

class SlugModel(object):
         slug = models.SlugField(unique=True)

         def save(self, *args, **kwargs):
         if not self.pk:
             words = self.name.split()
             filtered_words = [t for t in words if t.lower() not in STOP_WORDS]
             # unidecode() represents special characters (unicode data) in ASCII
             new_list = unidecode(' '.join(filtered_words))
             self.slug = slugify(new_list)[:50]
 
            super(SlugModel, self).save(*args, **kwargs)